### DISCUSSION TITLE
Fix ESLint error with interfaces in `global.d.ts`

### DISCUSSION THREAD

**Discussion by mikenikles** [sveltejs/kit#3743] [Q&A]
Hi :wave:,

I have the following interface in my `src/global.d.ts`:

```typescript
/// <reference types="@sveltejs/kit" />

interface User {
  login: string;
  name: string;
}
```

I use the `User` interface in a component like this: `const user: User = getUser()`.

When I run the `lint` npm script, it reports `error 'User' is not defined  no-undef`

Does anyone know how I can teach ESLint about the interfaces defined in `src/global.d.ts`?

**Comment by irvanherz** [comment]
Create a file named `.eslintignore`. Then add `src/global.d.ts` to it.

**Reply by justingolden21** [reply]
This simply has eslint ignore global.d.ts. He's trying to do the *opposite* of that. He WANTS eslint to know about global.d.ts.

I've read that you can add `/// <reference path="../../../global.d.ts" />` inside the `<script>` tag in your `.svelte` files (with appropriate number of `../` of course), but this doesn't work for me.

**Comment by craig-jennings** [comment]
Coming in 3 years later - did anyone find a solution to this?
